sql/core/src/main/scala/org/apache/spark/sql/jdbc/OracleDialect.scala

```diff
@@ -68,5 +69,13 @@ private case object OracleDialect extends JdbcDialect {
     case _ => None
   }
 
+  override def beforeFetch(connection: Connection, properties: Map[String, String]): Unit = {
+    // Set general date and timestamp format before query.
+    val stmt = connection.createStatement()
+    stmt.execute("alter session set NLS_DATE_FORMAT = 'YYYY-MM-DD'")
+    stmt.execute("alter session set NLS_TIMESTAMP_FORMAT = 'YYYY-MM-DD HH24:MI:SS.FF'")
```

Comment:
Let me explain it more clearly. ```alter session set NLS_TIMESTAMP_FORMAT = 'YYYY-MM-DD HH24:MI:SS.FF'``` This will change the timestamp format of the connection session. Based on my understanding, `${options.table}` still has to follow the format. Before this PR, `${options.table}` is using the default format. Thus, after your PR, users are unable to directly migrate their workloads to the new version of Spark